codecov_cli/services/upload/coverage_file_finder.py

```diff
@@ -163,26 +166,76 @@
 
 class CoverageFileFinder(object):
     def __init__(
-        self, project_root: Path = None, folders_to_ignore: typing.List[str] = None
+        self,
+        project_root: Path = None,
+        folders_to_ignore: typing.List[str] = None,
+        explicitly_listed_files: typing.List[Path] = None,
+        disable_search: bool = False,
     ):
         self.project_root = project_root or Path(os.getcwd())
         self.folders_to_ignore = folders_to_ignore or []
+        self.explicitly_listed_files = explicitly_listed_files or None
+        self.disable_search = disable_search
 
     def find_coverage_files(self) -> typing.List[UploadCollectionResultFile]:
-        regex_patterns_to_include = globs_to_regex(coverage_files_patterns)
         regex_patterns_to_exclude = globs_to_regex(coverage_files_excluded_patterns)
+        coverage_files_paths = []
+        user_coverage_files_paths = []
+        if self.explicitly_listed_files:
+            user_filenames_to_include = []
+            files_excluded_but_user_includes = []
+            for file in self.explicitly_listed_files:
+                user_filenames_to_include.append(file.name)
+                if regex_patterns_to_exclude.match(file.name):
+                    files_excluded_but_user_includes.append(str(file))
+            if files_excluded_but_user_includes:
+                logger.warning(
+                    "Some files being explicitly added are found in the list of excluded files for upload.",
+                    extra=dict(
+                        extra_log_attributes=dict(
+                            files=files_excluded_but_user_includes
+                        )
+                    ),
+                )
+            regex_patterns_to_include = globs_to_regex(user_filenames_to_include)
+            multipart_include_regex = globs_to_regex(
+                [str(path.resolve()) for path in self.explicitly_listed_files]
+            )
+            user_coverage_files_paths = search_files(
```

Comment:
Should we also log files that were listed by the user (explicated_listed_files) that were NOT found by this?

Reply:
good point